src/main/java/com/google/devtools/build/docgen/templates/be/make-variables.vm

```diff
@@ -17,105 +17,121 @@ title: Make Variables
 #if (!$singlePage)
 <div class="toc">
   <ul>
-    <li><a href="#make-var-substitution">"Make" variable subsitution</a></li>
+    <li><a href="#use">Use</a></li>
     <li><a href="#predefined_variables">Predefined variables</a></li>
-    <li><a href="#location">"$(location)" substitution</a></li>
+    <li><a href="#predefined_genrule_variables">Predefined genrule variables</a></li>
+    <li><a href="#predefined_label_variables">Predefined source/output path variables</a></li>
+    <li><a href="#custom_variables">Custom variables</a></li>
   </ul>
 </div>
 #end
 <p>
-  This section describes how to use a special class of built-in string variables
-  that are called the "Make" environment.
+  "Make" variables are a special class of expandable string variables available
+  to attributes marked as <i>"Subject to 'Make variable' substitution"</i>.
+</p>
+
+<p>
+  These can be used, for example, to inject specific toolchain paths into
+  user-constructed build actions.
+</p>
 
-  Defining custom "Make" variables is not supported.
+<p>
+  Bazel provides both <i>predefined</i> variables, which are available to all
+  rules, and <i>custom</i> variables, which are defined in dependency rules and
+  only available to parent rules that depend on them.
```

Comment:
"parent" is confusing here. Just remove it, I think "only available to rules that depend on them."

Reply:
Done.